packages/core/src/actions/smartAccount/internal/runMiddlewareStack.ts

```diff
@@ -73,11 +74,20 @@ export async function _runMiddlewareStack<
     throw new AccountNotFoundError();
   }
 
-  const { dummyPaymasterAndData, paymasterAndData } = bypassPaymasterAndData(
-    overrides
-  )
+  const {
+    dummyPaymasterAndData,
+    paymasterAndData,
+  }: Pick<
+    ClientMiddleware<TContext>,
+    "dummyPaymasterAndData" | "paymasterAndData"
+  > = bypassPaymasterAndData(overrides)
     ? {
-        dummyPaymasterAndData: noopMiddleware,
+        dummyPaymasterAndData: async (uo, overrides) => ({
+          ...uo,
+          ...("paymasterAndData" in overrides
+            ? { paymasterAndData: "0x" }
+            : { paymaster: undefined }),
```

Comment:
Actually, really sorry, but can we set all four paymaster fields to undefined in this case (`paymaster`, `paymasterData`, `paymasterVerificationGasLimit` and `paymasterPostOpGasLimit`)? A strict reading of the spec doesn't allow any of the other three fields to be set if `paymaster` is not. Although Rundler and other common bundlers accept this, ideally the SDK should adhere to the spec.

Reply:
nothing else should be set at this point actually, so I can just leave it out